README.md

```diff
@@ -117,6 +117,24 @@ const ENABLE_MULTI_STATEMENT = true
 })()
 ```
 
+Include orginal positions from where the statement was parsed
+```js
+const mysqlParser = require('@verycrazydog/mysql-parser')
+const splitResult = mysqlParser.splitIncludeSourceMap([
+  'delimiter $$',
+  'SELECT 1$$',
+  'delimiter ;',
+  'SELECT 2;'
+].join('\n'))
+// Print
+// [
+//   { stmt: "SELECT 1", start: 12, end: 23 },
+//   { stmt: "SELECT 2", start: 35, end: 45 }
```

Comment:
The indexes also need to update after bug fix.

Reply:
I ran this code with deno and the output is still the same:

```
> import * as mysqlParser from './src/index.ts'
undefined
> mysqlParser
[Module: null prototype] {
  split: [Function: split],
  splitIncludeSourceMap: [Function: splitIncludeSourceMap]
}
> const splitResult = mysqlParser.splitIncludeSourceMap([
  'delimiter $$',
  'SELECT 1$$',
  'delimiter ;',
  'SELECT 2;'
].join('\n'))
undefined
> console.log(splitResult)
[
  { stmt: "SELECT 1", start: 12, end: 23 },
  { stmt: "SELECT 2", start: 35, end: 45 }
]
undefined
```

What bugfix are you talking about?

Reply:
Maybe better to explain in code.

```js
const sql = [
  'delimiter $$',
  'SELECT 1$$',
  'delimiter ;',
  'SELECT 2;'
].join('\n')
// This print `false`, and I expect it print `true`
console.log(sql.substring(12, 23) === 'SELECT 1')
```

Reply:
I expect something different. I want the start and end to point out the full source that was parsed into the statement.

Simple example to understand what Im going for:

```js
> sql = "select 1\n;select 2;"
"select 1\n;select 2;"
> output = mysqlParser.splitIncludeSourceMap(sql)
[
  { stmt: "select 1", start: 0, end: 10 },
  { stmt: "select 2", start: 10, end: 19 }
]
> sql.substring(0, 10)
"select 1\n;"
```

In this case:

```sql
select 1
;
```

Is parsed into

```sql
select 1
```

But I want to know exactly the full source, including the intermediate `\n` between the stmt and the delimiter `;`

Reply:
I think it is important to have `stmt` equals `sourceInput.substring(start, end)`. I think that is the common expectation. If not, that should be at least properly documented. One options is to have 2 pairs of `start` and `end` indexes to cover the common expectation (matching substring) and your expectation (full source).

Let's have a conclusion before any code changes. What is your purpose of knowing the exactly full source? 🤔

Reply:
The purpose is to build a SQL editor and to know exactly which part of the original text is responsible for each stmt that is going to be executed in the database

Reply:
But without the full source, I think that can still archive your purpose of highlighting the parsed statements in editor? 🤔
Just that the new line and delimiter won't get highlighted.

Reply:
I understand if you don't want this functionality in your source code. I can maintain my fork and use it directly from there.

But what I implemented is exactly what I'm looking for. I don't want to impose anything on you since I'm going for some very specific behavior. I'm happy to close the PR and keep my fork as an alternative implementation.

I appreciate your feedback and the time you have taken to review this, I really don't take it for granted.

If you want I can make the implementation you suggest so we merge this PR, but I'll maintain my fork with the functionality I want.

In the end, what do you prefer?

1. Close the PR without merging.
2. Make the changes so that `stmt` equals `sourceInput.substring(start, end)` and merge the PR.

Reply:
Thanks for your time to address my feedback too. Since both of us has invested our time on this, I want to have the feature included. We just need to figure out a design that both of us agree on.

I am thinking of this solution:
- Add a new boolean option to specify trim behavior in output
- Add a new boolean option to specify whether to retain delimiter in output
- Ensure `stmt` equals `sourceInput.substring(start, end)`

However that will result in `stmt` using full source including delimiter. Does that work for you? If that doesn't work or you need to process the `stmt` to remove delimiter, I can think of another solution.

Reply:
Thank you for taking the time to discuss this and for considering ways to include the feature. I really appreciate the effort.

After some thought, I’ve decided to stick with my fork, as my implementation fully meets my needs, and I’m not looking to further develop an alternative approach at this time. Given that, I think it makes the most sense to close the PR.

Again, I truly appreciate your feedback and the time you've spent reviewing this. Thanks for the discussion, and I look forward to seeing how the project evolves!

Reply:
No problem. Maintaining your own fork do make sense too so you can have greater control. In case someone else request similar feature in the future, we can work on that base on this PR 👍